### PR TITLE
Feature/integration with backend

### PR DESCRIPTION
**Pull request by olya-s**

### CODE REVIEW

File: src/components/history-container/history-container.jsx [hunk 1]
```diff
@@ -36,6 +38,12 @@ function HistoryContainer({ mode, setMode }) {
   const handleClick = (event) => {
     setMode('wait');
     setMessage(event.target.textContent);
+    let answer = event.target.textContent.toUpperCase();
+
+    if (answer === "DON'T KNOW") {
```
Comment (MykhailoHoroshko) [inline src/components/history-container/history-container.jsx]: use answer type for comparison

File: src/components/history-container/history-container.jsx [hunk 2]
```diff
@@ -36,6 +38,12 @@ function HistoryContainer({ mode, setMode }) {
...
+      answer = 'NOT_SURE';
+    }
+    answerQuestion('Player 1', 1, answer);
```
Comment (MykhailoHoroshko) [inline src/components/history-container/history-container.jsx]: player name mustn't be hard coded

File: src/screens/lobby/lobby.jsx [hunk 3]
```diff
@@ -4,38 +4,86 @@ import LeaveGameModal from '../../components/modals/leave-game/leave-game';
...
 function Lobby() {
+  const { gameData, setGameData, playerId } = useContext(GameDataContext);
   const [showLeaveModal, setLeaveModalActive] = useState(false);
   const [showSuggestModal, setSuggestModal] = useState(false);
+  const [suggestBtn, setSuggestBtn] = useState(true);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (!gameData.data.status) navigate('/');
+  });
```
Comment (MykhailoHoroshko) [inline src/screens/lobby/lobby.jsx]: add dependencies array

File: src/screens/lobby/lobby.jsx [hunk 4]
```diff
@@ -4,38 +4,86 @@ import LeaveGameModal from '../../components/modals/leave-game/leave-game';
...
+    if (gameData.data.status === PROCESSING_QUESTION) {
+      navigate('/play');
+    }
+  }, [gameData, navigate]);
+
+  const players = gameData.data.players;
+  const currentPlayer =
+    players && players.find((player) => player.player.name === playerId);
+  const playersWithoutYou =
+    players && players.filter((player) => player.player.name !== playerId);
```
Comment (MykhailoHoroshko) [inline src/screens/lobby/lobby.jsx:43]: useMemo

File: src/screens/lobby/lobby.jsx [hunk 5]
```diff
@@ -4,38 +4,86 @@ import LeaveGameModal from '../../components/modals/leave-game/leave-game';
...
           <div className="input-screen__player-card-wrapper">
-            {[1, 2, 3, 4].map((num) => (
+            {currentPlayer && (
               <PlayerCard
-                key={num}
-                avatarClassName={`avatar0${num}`}
-                name={`Player ${num}`}
-                playerStausClassName={'no'}
+                avatarClassName={`avatar0${1}`}
+                name={`Player ${1}`}
```
Comment (MykhailoHoroshko) [inline src/screens/lobby/lobby.jsx]:
```suggestion
                avatarClassName="avatar01"
                name={currentPlayer.name}
```

File: src/screens/lobby/lobby.jsx [hunk 6]
```diff
@@ -4,38 +4,86 @@ import LeaveGameModal from '../../components/modals/leave-game/leave-game';
...
+                playerStatusClassName={
+                  currentPlayer.state === 'NOT_READY' ? 'no' : 'yes'
```
Comment (MykhailoHoroshko) [inline src/screens/lobby/lobby.jsx]: add states, answers to constants

File: src/screens/lobby/lobby.jsx [hunk 7]
```diff
@@ -4,38 +4,86 @@ import LeaveGameModal from '../../components/modals/leave-game/leave-game';
...
                 isYou
               />
-            ))}
+            )}
+            {playersWithoutYou ? (
+              playersWithoutYou.map((player, index) => (
+                <PlayerCard
+                  key={player.player.name}
+                  avatarClassName={`avatar0${index + 2}`}
+                  name={`Player ${index + 2}`}
```
Comment (MykhailoHoroshko) [inline src/screens/lobby/lobby.jsx:68]: maybe `player.name` ?

**Review by MykhailoHoroshko — CHANGES_REQUESTED** [review]

File: src/screens/lobby/lobby.jsx (hunk 7 above)
Comment (olya-s) [reply src/screens/lobby/lobby.jsx:68]: maybe, but now it's just an id and looks not good. For now we have not possibility to get normal name. So should I replace this with player.name anyway?

Comment (MykhailoHoroshko) [reply src/screens/lobby/lobby.jsx:68]: ok, skip for now

File: src/components/answer-form/answer-form.jsx [hunk 8]
```diff
@@ -3,41 +3,62 @@ import { ReactComponent as Check } from '../../assets/svg/check-icon-btn.svg';
...
 
   if (props.mode === 'answer') {
     btnRow = (
-      <div className="row" onClick={props.onClick}>
-        <Btn className="btn-green-solid btn-third">
+      <form className="row">
+        <Btn
+          className="btn-green-solid btn-third"
+          value={YES}
+          onClick={props.onClick}
```
Comment (MykhailoHoroshko) [inline src/components/answer-form/answer-form.jsx:17]:
```suggestion
          onClick={() => props.onClick(e, YES)}
```

File: src/components/history-container/history-container.jsx [hunk 9]
```diff
@@ -27,15 +28,19 @@ function HistoryContainer({ mode, setMode }) {
...
     if (currentQuestion !== '') {
       history.push({ user: currentUser, question: currentQuestion });
+      // askQuestion(playerId, gameId, currentQuestion);
 
       setCurrentQuestion('');
       setDisabled(true);
     }
   };
 
   const handleClick = (event) => {
```
Comment (MykhailoHoroshko) [inline src/components/history-container/history-container.jsx:38]:
```suggestion
  const handleClick = (event, answer) => {
```

File: src/components/history-container/history-container.jsx [hunk 10]
```diff
@@ -27,15 +28,19 @@ function HistoryContainer({ mode, setMode }) {
...
+    event.preventDefault();
     setMode('wait');
-    setMessage(event.target.textContent);
+    setMessage(event.target.value);
```
Comment (MykhailoHoroshko) [inline src/components/history-container/history-container.jsx:41]:
```suggestion
    setMessage(answer);
```

File: src/screens/lobby/lobby.jsx [hunk 11]
```diff
@@ -4,38 +4,84 @@ import LeaveGameModal from '../../components/modals/leave-game/leave-game';
...
+                  playerStatusClassName={
+                    player.state === 'NOT_READY' ? 'no' : 'yes'
```
Comment (MykhailoHoroshko) [inline src/screens/lobby/lobby.jsx]: NOT_READY from constants

**Review by MykhailoHoroshko — APPROVED** [review]

**Pull request merged**